### ISSUE TITLE
Documentation to include delete operation examples

### ISSUE THREAD

**Issue by brianburnett** [lni/dragonboat#182]
> Note: for reported bugs, please fill in the following details. bug reports without detailed steps on how to reproduce will be automatically closed. 

### Dragonboat version
v.3.3.4

### Expected behavior
The code examples and/or go doc should describe the details to properly remove entries from Raft logs and the underlying persistence stores.

### Actual behavior
Example code and go doc demonstrate the addition of entries but it is unclear on how to perform deletions.

### Steps to reproduce the behavior
Request is relating to documentation, specifically for the OnDiskStateMachine.

**Comment by lni** [comment]
@brianburnett In your email, you mentioned that 
> However, for my purposes I need to be able to support ephemeral keys and expire entries after a period of time.

My understanding is you want some of your data entries in OnDiskStateMachine to be expired after a period of time. 

Raft log is an append only log of all operations you proposed for your OnDiskStateMachine, it can be compacted meaning you can request to have no longer required log entries (those already applied into your OnDiskStateMachine) to be removed, but there is no way to remove certain selected Raft log entries. 

To support the feature you mentioned, you need to make a proposal to request your OnDiskStateMachine to ignore and remove all entries that carry certain timestamp. Your OnDiskStateMachine code need to implement such logic. 

Please have a look at the section 2 of the [Raft paper](https://raft.github.io/raft.pdf), it explains how Raft logs and OnDiskStateMachine work together as the Replicated State Machine.

**Comment by brianburnett** [comment]
Yes, I needed to better understand how to manage the removal of entries before opening up a ticket on implementing a mechanism to expire them. That feature would be similar to other Raft implementations such as RedisRaft's Expire or etcd's Lease.

For this issue I was hoping to see if it was possible to add some sample code that follows your existing examples.

Such as with your ondisk example code.  If the commands (GET, PUT) could be extended to include DELETE, what would be the proper way to call nh.SyncPropose() on an existing key?

```
                                if rt == PUT {
					kv := &KVData{
						Key: key,
						Val: val,
					}
					data, err := json.Marshal(kv)
					if err != nil {
						panic(err)
					}
					_, err = nh.SyncPropose(ctx, cs, data)
					if err != nil {
						fmt.Fprintf(os.Stderr, "SyncPropose returned error %v\n", err)
					}
				} else if rt == DELETE {
                                       // key exists from prior PUT
                                       ...
                                        _, err = nh.SyncPropose(ctx, cs, ???)
                                       ...
                                } else {
					result, err := nh.SyncRead(ctx, exampleClusterID, []byte(key))
					if err != nil {
						fmt.Fprintf(os.Stderr, "SyncRead returned error %v\n", err)
					} else {
						fmt.Fprintf(os.Stdout, "query key: %s, result: %s\n", key, result)
					}
				}```

Thanks!

**Comment by lni** [comment]
For the replication state machine (RSM) model, PUT and DELETE are nothing different from each other, they both require changing the state of the RSM and thus they should be implemented as proposals. 

I'd like to keep the examples as simple as possible, maybe I can add some comments in the example to indicate that PUT and DELETE should both be implemented as proposals. 

Please feel free to open a new issue for the auto expiring feature.
